src/constructs/core/ssm.ts

```diff
@@ -0,0 +1,97 @@
+import { readFileSync } from "fs";
+import { join } from "path";
+import type { IGrantable, IPrincipal } from "@aws-cdk/aws-iam";
+import { Policy, PolicyStatement } from "@aws-cdk/aws-iam";
+import { Code, Runtime, SingletonFunction } from "@aws-cdk/aws-lambda";
+import type { Reference } from "@aws-cdk/core";
+import { Construct, CustomResource, Duration } from "@aws-cdk/core";
+import { AwsCustomResourcePolicy } from "@aws-cdk/custom-resources";
+import type { CustomResourceGetParameterProps } from "./custom-resources/interfaces";
+import type { AppIdentity } from "./identity";
+import type { GuStack } from "./stack";
+
+export interface GuSSMParameterProps {
+  parameter: string;
+  secure?: boolean;
+}
+
+export interface GuSSMIdentityParameterProps extends GuSSMParameterProps, AppIdentity {}
+
+const stripped = (str: string) => str.replace(/[-/]/g, "");
+
+export class GuSSMParameter extends Construct implements IGrantable {
+  private readonly customResource: CustomResource;
+  readonly grantPrincipal: IPrincipal;
+
+  // eslint-disable-next-line custom-rules/valid-constructors -- TODO: Remove once linting rules have been relaxed for this
+  constructor(scope: GuStack, props: GuSSMParameterProps) {
```

Comment:
https://github.com/guardian/cdk/pull/352 allows for this constructor 🎉  a rebase of `main` should allow you to remove this line.

Reply:
Nice one, done.